src/ln/peer_handler.rs

```diff
@@ -113,15 +113,14 @@ pub struct PeerManager<Descriptor: SocketDescriptor> {
 }
 
 macro_rules! encode_msg {
-	($msg: expr, $msg_code: expr) => {
-		{
-			let just_msg = $msg.encode();
-			let mut encoded_msg = Vec::with_capacity(just_msg.len() + 2);
-			encoded_msg.extend_from_slice(&byte_utils::be16_to_array($msg_code));
-			encoded_msg.extend_from_slice(&just_msg[..]);
-			encoded_msg
-		}
-	}
+	($msg: expr, $msg_code: expr) => {{
+		let mut w = Writer::new(::std::io::Cursor::new(vec![]));
+		0u16.write(&mut w).unwrap();
+		$msg.write(&mut w).unwrap();
+		let mut msg = w.into_inner().into_inner();
+		byte_utils::be16_to_array_inplace(msg.len() as u16-2, &mut msg[..2]);
```

Comment:
You could just msg[..2].copy_from_slice(byte_utils::be16_to_array(msg.len() - 2)) to avoid adding another function. It should optimize away pretty easily.